### DISCUSSION TITLE
Rustdesk activating discrete GPU even when set to interated/power savings

### DISCUSSION THREAD

**Discussion by nickyivyca** [rustdesk/rustdesk#6753] [General]
### Bug Description

I have an XPS 15 9560 with an Intel 630 and Nvidia 1050M. I have the Nvidia driver set to select the Integrated GPU, but I still see the Nvidia GPU activated by RustDesk whenever I use it. I also see this on a Precision 5520 which has a similar setup except with a Quadro M1200. 

![image](https://github.com/rustdesk/rustdesk/assets/3387157/5d368099-1844-4b7e-9da9-09294544ba11)

I have the integrated GPU set in both Nvidia Control Panel as well as the Windows graphics settings, both for rustdesk.exe and RuntimeBroker_rustdesk.exe.

![image](https://github.com/rustdesk/rustdesk/assets/3387157/b05ee960-a361-4890-b327-c1459095afa9)
![image](https://github.com/rustdesk/rustdesk/assets/3387157/f03e2886-ac7e-4486-9f19-15796fc71ded)

Strangely enough, however, Task Manager still shows GPU 0 under the rustdesk listing. 
![image](https://github.com/rustdesk/rustdesk/assets/3387157/1d722581-f032-4842-a360-5338cfe8447f)
But, I do see a _slight_ blip in GPU 1 under the usage when I end the Rustdesk connection, and the memory usage goes from close to 0 to 0.
![image](https://github.com/rustdesk/rustdesk/assets/3387157/dd1f493c-c5bb-4cd9-b9ac-707d8d487735)

I've tried a couple different driver versions and it always has behaved the same way.

### How to Reproduce

Have a laptop with a dedicated Nvidia GPU, set Rustdesk to use integrated graphics, open a Rustdesk session, and see in the Nvidia indication that the GPU is still running.

The GPU does not enable until I begin the session (when the screen with password entry pops up) but does not shut off until I close the Rustdesk home window.

### Expected Behavior

Rustdesk should not cause the discrete graphics to activate at any time, given that I've set the driver settings to use the integrated graphics.

### Operating system(s) on local side and remote side

Win10 -> Win10 or Ubuntu

### RustDesk Version(s) on local side and remote side

1.2.3 -> 1.2.3

### Screenshots

![image](https://github.com/rustdesk/rustdesk/assets/3387157/5d368099-1844-4b7e-9da9-09294544ba11)

**Comment by 21pages** [comment]
The NVIDIA Control Panel can determine which GPU to use for rendering and capturing, but it cannot determine which GPU to use for encoding and decoding. During program startup, the availability of encoding and decoding capabilities for each GPU is checked. Each GPU may be briefly utilized for this purpose, but I didn't reproduce the slight blip in GPU 1 on disconnection, the GPU1 utilization rate remains at 0.

**Reply by nickyivyca** [reply]
Yeah I was able to prevent this from happening by disabling hardware codec (I thought I tried this as part of the initial debug but I think I had to restart Rustdesk for the setting to fully take it seems). It is hard to tell if the GPU is indeed pulling any extra power when it is 'activated' to your point of the GPU just being polled.

**Comment by busFred** [comment]
I am also being affected by this issue, but i am on linux. whenever i remote desktop to other machine, the session is rendered on nvidia gpu but not on the built-in integrated graphics. is there a way to set which gpu to use from within rustdesk? i don't think linux has nvidia control panel
